src/Models/Request/PointsBatch.php

```diff
@@ -25,15 +27,43 @@ public function __construct()
 
     public function addPoint(PointStruct $point): void
     {
-        $pointAr = $point->toArray();
-
-        $this->batchPoints['ids'][] = $pointAr['id'];
-        $this->batchPoints['vectors'][] = $pointAr['vector'];
-        $this->batchPoints['payloads'][] = $pointAr['payload'] ?: null;
+        $this->batchPoints['ids'][] = $point->getId();
+        $this->batchPoints['vectors'][] = $point->getVector();
+        $this->batchPoints['payloads'][] = $point->getPayload();
     }
 
     public function toArray(): array
     {
+        if (
+            count($this->batchPoints['ids']) === 0 ||
+            count($this->batchPoints['vectors']) !== count($this->batchPoints['ids'])
+        ) {
+            throw new InvalidArgumentException('You need to add at least one point and the number of id should match with vectors.!');
+        }
+        $isNamed = $this->batchPoints['vectors'][0]->isNamed();
+
+        $vectors = [];
+        $numberOfVectors = count($this->batchPoints['vectors']);
+        if ($isNamed) {
+            /** @var VectorStruct $vector */
+            foreach ($this->batchPoints['vectors'] as $index => $vector) {
+                $vectorArr = $vector->toArray();
+                foreach ($vectorArr as $name => $item) {
+                    if (!isset($vectors[$name])) {
+                        $vectors[$name] = array_fill(0, $numberOfVectors, null);
+                    }
+                    $vectors[$name][$index] = $item;
+                }
+            }
+        } else {
+            /** @var VectorStruct $vector */
+            foreach ($this->batchPoints['vectors'] as $vector) {
+                $vectors[] = $vector->toArray();
+            }
+        }
+
+        $this->batchPoints['vectors'] = $vectors;
+
```

Comment:
I don't like doing this, but I could not find any other way. I am open to any suggestions.

Reply:
Maybe after initializing VectorStructInterface, we can implement VectorStruct, NamedVectorStruct, and MultiVectorStruct separately. But I don't want to change everything so much, too, for the time being.